src/main/java/Duke.java

```diff
@@ -1,10 +1,271 @@
+        String textFilePath = "C:\\Users\\cwxky\\projects\\cs2113-git\\data\\duke.txt";
+        File textFile = new File(textFilePath);
+        try {
+            if (!textFile.exists()) {
+                textFile.createNewFile();
+                System.out.println("File has been created!");
+            } else {
+                System.out.println("File already exists.");
+            }
+        } catch (IOException e) {
+            System.out.println("Duke has failed to make the file");
+        }
+
+        String line;
+        ArrayList<Task> tasksList = new ArrayList<Task>();
+        ArrayList<String> taskOverrideList = new ArrayList<String>();
+        int count = 0;
+        Scanner in = new Scanner(System.in);
+        System.out.println("Duke: Hello! I'm Duke\n" + "Duke: What can I do for you?");
+
+        while (true) {
+            line = in.nextLine();
+            String[] words = line.split(" ");
+            switch (words[0]) {
+            case COMMAND_BYE:
+                System.out.println(MESSAGE_BYE);
+                System.exit(0);
+                break;
+            case COMMAND_LIST:
+                try {
+                    if (tasksList.isEmpty()) {
+                        throw new EmptyListException(line);
+                    } else {
+                        System.out.println("Here are the tasks for today: ");
+                        for (int i = 0; i < tasksList.size(); i++) {
+                            System.out.println(tasksList.get(i));
+                        }
+                    }
+                    printFileContents(textFilePath);
+                } catch (EmptyListException e) {
+                    System.out.println(e.getExceptionMessage());
+                } catch (FileNotFoundException f) {
+                    System.out.println("File not found.");
+                }
+                break;
+            case COMMAND_MARK:
+                int itemNumber = Integer.parseInt(line.split(" ")[1]) - 1;
+                try {
+                    if (itemNumber < 0 || itemNumber > tasksList.size() - 1) {
+                        throw new IllegalTaskNumber("this task does not exist");
+                    } else {
+                        Task toBeChanged = tasksList.get(itemNumber);
+                        if (toBeChanged.isDone) {
+                            throw new IllegalMarkingException("already marked");
+                        } else {
+                            toBeChanged.markAsDone(toBeChanged);
+                            String textToAdd = toBeChanged.toFileString();
+                            taskOverrideList.set(itemNumber, textToAdd);
+                            writeToFile(textFilePath, taskOverrideList.get(0));
+                            for (int i = 1; i < taskOverrideList.size(); i++) {
+                                appendToFile(textFilePath, taskOverrideList.get(i));
+                            }
+                        }
+                    }
```

Comment:
For this block of code, perhaps you may want to consider further abstracting it out through something like a helper function? The motivation would be to reduce deeply nested code and improve code readibility.